File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func levelOrder(root *TreeNode) [][]int {
	if root == nil {
		return [][]int{}
	}
	
	curr := []*TreeNode{root}
	children := []*TreeNode{}
	out := [][]int{}

	for len(curr) > 0 {
		children = []*TreeNode{}
		values := []int{}
		for i := range curr {
			node := curr[i]
			values = append(values, node.Val)
			if node.Left != nil {
				children = append(children, node.Left)
			}

			if node.Right != nil {
				children = append(children, node.Right)
			}
		}
		out = append(out, values)
		curr = children
	}
	return out
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kthSmallest(root, k) {
        const sorted = this.walk(root)
        return sorted[k - 1]
    }
    walk(root, path=[]) {
        if (root.left) {
            this.walk(root.left, path)
        }

        path.push(root.val)

        if (root.right) {
            this.walk(root.right, path)
        }
        return path
    }
}
